### PR TITLE
initial commit of stride range and a couple utils

### PR DESCRIPTION
**Pull request by sccolbert**
Needs:
1. Implementations of `back()` and `setBack()` which drop extra elements modulo the step size.
2. Not sure if I like the name `utils.ts`. Thing of renaming to `drop.ts`.

### CODE REVIEW

File: src/range/stride.ts [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+   *
+   * @returns The value at the front of the range.
+   *
+   * #### Notes
+   * This returns the front of the source range.
+   *
+   * If the range is empty, the behavior is undefined.
+   */
+  front(): T {
+    return this.source.front();
```
Comment (blink1073) [inline src/range/stride.ts]: Shouldn't this also step after grabbing the front?

File: src/range/stride.ts [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+   *
+   * @param value - The value to set at the front of the range.
+   *
+   * #### Notes
+   * This sets the value at the front of the source range.
+   *
+   * If the range is empty, the behavior is undefined.
+   */
+  setFront(value: T): void {
+    this.source.setFront(value);
```
Comment (blink1073) [inline src/range/stride.ts]: Again, shouldn't this drop some values depending on the step?

**Comment by blink1073** [comment]
I'm fine with `utils`, if they are not intended to be used directly.

File: src/range/stride.ts (hunk 1 above)
Comment (sccolbert) [inline src/range/stride.ts]: It steps on `dropFront()`.

Iterating a range takes the form:

```typescript
while (!range.isEmpty()) {
  doSomethingWith(range.front());
  range.dropFront();
}
```

i.e. `front()` can be accessed multiple times before stepping (if needed)

File: src/range/stride.ts (hunk 2 above)
Comment (sccolbert) [inline src/range/stride.ts]: See answer above.

**Comment by sccolbert** [comment]
The `dropFront(range, n)` and `dropBack(range, n)` functions are generally useful, and I would expect them to be used externally. I think you actually asked for these functions early on :)

**Comment by blink1073** [comment]
Fair enough, I agree with code as-is, and with the name `drop.ts`.

**Pull request merged**